src/main/java/org/broadinstitute/hellbender/utils/io/TextFeatureReader.java

```diff
@@ -0,0 +1,244 @@
+package org.broadinstitute.hellbender.utils.io;
+
+import htsjdk.samtools.util.BlockCompressedInputStream;
+import htsjdk.tribble.*;
+import htsjdk.tribble.index.Block;
+import htsjdk.tribble.index.Index;
+import org.broadinstitute.hellbender.engine.GATKPath;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.IndexUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.*;
+import java.nio.channels.Channels;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+public final class TextFeatureReader<F extends Feature> implements FeatureReader<F> {
```

Comment:
Could you add some javadoc explaining this one?

Reply:
I'm not totally clear on why this instead of the existing infrastructure.  This is nicer and uses the GATK types, but it introduces confusing differences from the behavior of the existing readers. For instance it doesn't use the cloud wrappers which I suspect will lead to performance issues if you try to use it on a file in gcp.
It also doesn't respect non-pathable GATKPaths like stream constructed ones, so using that type doesn't really add anything.  

It's fine to add it, but it adds another layer of complexity if we ever move to the new htsjdk input APIs.

Reply:
I would probably expose the option to explicitly set the index paths instead of relying on the magic index finding.  The old stuff doesn't but that was a mistake which we're trying to stop replicating.